### ISSUE TITLE
On macOS, file paths in config file don't expand "~"

### ISSUE THREAD

**Issue by jeresch** [mapcrafter/mapcrafter#233]
Instead, the program will create a new "~" directory in the current working directory.

**Comment by jeresch** [comment]
It appears that filenames are fed directly into the initializer for std::ifstream (see line 185 in src/mapcraftercore/config/iniconfig.cpp).  According to [here](http://www.cplusplus.com/reference/fstream/ifstream/ifstream/),  this depends on library implementation.

I propose a quick parsing of the filenames before creating an ifstream, to catch "~" characters and expand them.  I managed to delete half of my home directory by accident

**Comment by m0r13** [comment]
You are right. It would be just a quick replacement of that tilde character in the filename, that shouldn't be a big problem. I think there is even a posix function for it.

What do you mean with the inline spaces? Is that as expected for you? I don't think that one should have to use backslashes for spaces in the config file as the spaces don't have another meaning there.

**Comment by jeresch** [comment]
I agree.  Spaces should be fine.

 I see that the boost filesystem library provides some functionality in the `path` class to expand file paths in a system-dependent way.  I'm going to try and do that and see if I can get it to work.  I think the best place to expand would be within the few loadFile and writeFile methods, but it could be done upstream at the parsing stage.

If this isn't too critical an issue, I'd love to try my hand at it and report back soon.

**Comment by m0r13** [comment]
Sure, you are welcome to try :)

**Comment by m0r13** [comment]
I think it would be best to be done at the parsing stage, like [making the paths relative to the configuration file](https://github.com/mapcrafter/mapcrafter/blob/master/src/mapcraftercore/config/mapcrafterconfig.cpp#L104).

**Comment by jeresch** [comment]
Cool, I'll see what I can do.  Thanks!
